### PR TITLE
[TIMOB-26014] Ti.Media video recording is not working

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-26014](https://jira.appcelerator.org/browse/TIMOB-26014)

Fix Ti.Media video recording. This PR also includes clean-up codebase for Windows 8.1 support.

Note: 

- `picturesLibrary` capability is required in order to save a photo onto Windows Pictures Library folder.
- `videosLibrary` capability is required in order to save a video onto Windows Videos Library folder.
For more information about photo/video capabilities see [Camera and Photo Gallery APIs](https://docs.appcelerator.com/platform/latest/#!/guide/Camera_and_Photo_Gallery_APIs-section-src-29004913_CameraandPhotoGalleryAPIs-Grantaccesstopictureslibraryy).

## Video recording with default UI

- Expected: Recorded video is saved onto Video Library folder

```js
var win = Ti.UI.createWindow({
    backgroundColor: 'green'
});

var btnTakeMovie = Ti.UI.createButton({ title: 'Take a Video', top: 10, left: 5 });

win.add(btnTakeMovie)

btnTakeMovie.addEventListener('click', function (e) {
    Titanium.Media.showCamera({
        success: function (e) {
            Ti.API.info('success: ' + e.media.nativePath);
            Ti.API.info('         ' + e.media.size);
        },
        cancel: function (e) {
            Ti.API.info('cancel: ' + JSON.stringify(e));
        },
        error: function (e) {
            Ti.API.info('error: ' + JSON.stringify(e));
        },
        saveToPhotoGallery: true,
        mediaTypes: [Titanium.Media.MEDIA_TYPE_VIDEO],
        videoMaximumDuration: 1000000,
        videoQuality: Titanium.Media.QUALITY_HIGH
    });
});

win.open();
```

## Video recording with custom UI

- Expected: Recorded video is saved onto Video Library folder

```js
var win = Ti.UI.createWindow({
    backgroundColor: 'green'
});

var btnTakeMovie = Ti.UI.createButton({ title: 'Take a Video', top: 10, left: 5 });

win.add(btnTakeMovie)

var overlay = Ti.UI.createView({
    layout: 'vertical',
    height: '20%',
    width: Ti.UI.FILL,
    bottom: 0
}),
    startButton = Ti.UI.createButton({
        title: 'START RECORDING',
        backgroundColor: 'red',
        touchEnabled: true
    }),
    stopButton = Ti.UI.createButton({
        title: 'STOP RECORDING',
        backgroundColor: 'red',
        touchEnabled: false
    });

startButton.addEventListener('click', function () {
    Ti.Media.startVideoCapture();
    Ti.API.info('STARTED');
    startButton.touchEnabled = false;
    stopButton.touchEnabled  = true;
});

stopButton.addEventListener('click', function () {
    Ti.Media.stopVideoCapture();
    Ti.API.info('STARTED');
    startButton.touchEnabled = true;
    stopButton.touchEnabled  = false;
});

overlay.add(startButton);
overlay.add(stopButton);

btnTakeMovie.addEventListener('click', function (e) {
    Titanium.Media.showCamera({
        overlay: overlay,
        success: function (e) {
            Ti.API.info('success: ' + e.media.nativePath);
            Ti.API.info('         ' + e.media.size);
        },
        cancel: function (e) {
            Ti.API.info('cancel: ' + JSON.stringify(e));
        },
        error: function (e) {
            Ti.API.info('error: ' + JSON.stringify(e));
        },
        saveToPhotoGallery: true,
        mediaTypes: [Titanium.Media.MEDIA_TYPE_VIDEO],
        videoMaximumDuration: 1000000,
        videoQuality: Titanium.Media.QUALITY_HIGH
    });
});

win.open();
```

Also make sure nothing is broken on taking pictures;

## Take a Photo with default UI

- Expected: Photo is saved onto Pictures Library folder

```js
var win = Ti.UI.createWindow({
    backgroundColor: 'green'
});

var btnTakeMovie = Ti.UI.createButton({ title: 'Take a Photo', top: 10, left: 5 });

win.add(btnTakeMovie)

btnTakeMovie.addEventListener('click', function (e) {
    Titanium.Media.showCamera({
        success: function (e) {
            Ti.API.info('success: ' + e.media.nativePath);
            Ti.API.info('         ' + e.media.size);
        },
        cancel: function (e) {
            Ti.API.info('cancel: ' + JSON.stringify(e));
        },
        error: function (e) {
            Ti.API.info('error: ' + JSON.stringify(e));
        },
        saveToPhotoGallery: true,
        mediaTypes: [Titanium.Media.MEDIA_TYPE_PHOTO],
    });
});

win.open();
```

## Take a photo with custom UI

- Expected: Photo is saved onto Pictures Library folder

```js
var win = Ti.UI.createWindow({
    backgroundColor: 'green'
});

var btnTakeMovie = Ti.UI.createButton({ title: 'Take a Photo', top: 10, left: 5 });

win.add(btnTakeMovie)

var overlay = Ti.UI.createView({
    layout: 'vertical',
    height: '20%',
    width: Ti.UI.FILL,
    bottom: 0
}),
    takeButton = Ti.UI.createButton({
        title: 'TAKE A SHOT',
        backgroundColor: 'red',
    }),
    hideButton = Ti.UI.createButton({
        title: 'HIDE PREVIEW',
        backgroundColor: 'red',
    });

takeButton.addEventListener('click', function () {
    Ti.Media.takePicture();
    Ti.API.info('TAKE');
});

hideButton.addEventListener('click', function () {
    Ti.Media.hideCamera();
    Ti.API.info('HIDE');
});

overlay.add(takeButton);
overlay.add(hideButton);

btnTakeMovie.addEventListener('click', function (e) {
    Titanium.Media.showCamera({
        overlay: overlay,
        success: function (e) {
            Ti.API.info('success: ' + e.media.nativePath);
            Ti.API.info('         ' + e.media.size);
        },
        cancel: function (e) {
            Ti.API.info('cancel: ' + JSON.stringify(e));
        },
        error: function (e) {
            Ti.API.info('error: ' + JSON.stringify(e));
        },
        saveToPhotoGallery: true,
        mediaTypes: [Titanium.Media.MEDIA_TYPE_PHOTO],
    });
});

win.open();
```

### CODE REVIEW

File: Source/Media/src/Media.cpp [hunk 1]
```diff
@@ -539,32 +517,50 @@ namespace TitaniumWindows
...
+				cameraCaptureUI->VideoSettings->MaxResolution = CameraCaptureUIMaxVideoResolution::HighDefinition;
+			} else if (options.videoQuality == Titanium::Media::Quality::Medium) {
+				cameraCaptureUI->VideoSettings->MaxResolution = CameraCaptureUIMaxVideoResolution::StandardDefinition;
+			} else if (options.videoQuality == Titanium::Media::Quality::Low) {
+				cameraCaptureUI->VideoSettings->MaxResolution = CameraCaptureUIMaxVideoResolution::LowDefinition;
+			}
+			captureMode = CameraCaptureUIMode::Video;
+		} else {
+			// TODO: Provide a option to specify aspect ratio
+			cameraCaptureUI->PhotoSettings->CroppedAspectRatio = Size(16, 9);
```
Comment (garymathews) [inline Source/Media/src/Media.cpp]: Should we default Photos to a `4:3` ratio? This is usually the ratio photos are taken with as it matches the ratio of the sensor

Comment (infosia) [reply Source/Media/src/Media.cpp]: Makes sense, updated to `4:3`. I also wanted to make it more customizable, added `photoAspectRatio` configuration property like below but I noticed Windows' `CroppedAspectRatio` in `PhotoSetting` is basically ignored in `CameraCaptureUI`. I left this feature anyway.

```js
Titanium.Media.showCamera({
    saveToPhotoGallery: true,
    mediaTypes: [Titanium.Media.MEDIA_TYPE_PHOTO],
    photoAspectRatio: { width: 3, height: 4 }
});
```

**Comment by ssjsamir** [comment]
@infosia The only test cases i am able to verify is the 2nd and last one. (Recording and taking pictures with a custom UI). The first test case I am unable to see a video in the videos library and the 3rd test case throws an error on device.

**Test Environment**
```
Name = Microsoft Windows 10 Enterprise
Version = 10.0.15063
Architecture = 64bit
CPU Cores = 8
Memory = 15.9GB
Node.js = 8.9.1
Appcelerator CLI = 7.0.6
```

**Comment by infosia** [comment]
@ssjsamir Thanks fixed it, I was able to make it running locally. For the `Capabilities` you need `<manifest>` tag before `Capabilities` such as

```xml
<windows>
   <manifest>
      <Capabilities>
        <DeviceCapability Name="microphone"/>
        <DeviceCapability Name="webcam"/>
        <Capability Name="picturesLibrary"/>
        <Capability Name="videosLibrary"/>
    </Capabilities>
  </manifest>
</windows>
```

**Pull request merged**